crane-parse/src/lib.rs

```diff
@@ -338,57 +338,100 @@ fn expr<'src>() -> impl ChumskyParser<'src, ParserStream<'src>, NodeId, ParserEx
+            op.then(additive).repeated(),
+            |lhs, (op, rhs), state: &mut ParserState| {
                 state
                     .current_unit_mut()
                     .new_expr(Expr::BinaryOp { lhs, op, rhs })
             },
         );
 
-        let equality = cmp!(Equal)
-            .map(|_| BinaryOp::Eq)
-            .or(cmp!(NotEqual).map(|_| BinaryOp::Neq));
-        let cmp_equality =
-            sum.clone()
-                .foldl_with_state(equality.then(sum).repeated(), |lhs, (op, rhs), state| {
-                    state
-                        .current_unit_mut()
-                        .new_expr(Expr::BinaryOp { lhs, op, rhs })
-                });
-
-        let relational = cmp!(LessThan)
+        let op = cmp!(LessThan)
             .map(|_| BinaryOp::Lt)
             .or(cmp!(LessThanOrEqual).map(|_| BinaryOp::Leq))
             .or(cmp!(GreaterThan).map(|_| BinaryOp::Gt))
             .or(cmp!(GreaterThanOrEqual).map(|_| BinaryOp::Geq));
-        let cmp_relational = cmp_equality.clone().foldl_with_state(
-            relational.then(cmp_equality).repeated(),
-            |lhs, (op, rhs), state| {
+        let relational = bitshift.clone().foldl_with_state(
+            op.then(bitshift).repeated(),
+            |lhs, (op, rhs), state: &mut ParserState| {
                 state
                     .current_unit_mut()
                     .new_expr(Expr::BinaryOp { lhs, op, rhs })
             },
         );
 
-        cmp_relational
+        let op = cmp!(Equal)
+            .map(|_| BinaryOp::Eq)
+            .or(cmp!(NotEqual).map(|_| BinaryOp::Neq));
+        let equality = relational.clone().foldl_with_state(
+            op.then(relational).repeated(),
+            |lhs, (op, rhs), state: &mut ParserState| {
+                state
+                    .current_unit_mut()
+                    .new_expr(Expr::BinaryOp { lhs, op, rhs })
+            },
+        );
+
+        let binop = |lhs: NodeId, (op, rhs), state: &mut ParserState| {
+            state
+                .current_unit_mut()
+                .new_expr(Expr::BinaryOp { lhs, op, rhs })
+        };
+        let op = bit!(And).map(|_| BinaryOp::BitwiseAnd);
+        let bitwise_and = equality
+            .clone()
+            .foldl_with_state(op.then(equality).repeated(), binop);
+
+        let op = bit!(Xor).map(|_| BinaryOp::Xor);
+        let bitwise_xor = bitwise_and
+            .clone()
+            .foldl_with_state(op.then(bitwise_and).repeated(), binop);
+
+        let op = bit!(Or).map(|_| BinaryOp::BitwiseOr);
+        let bitwise_or = bitwise_xor
+            .clone()
+            .foldl_with_state(op.then(bitwise_xor).repeated(), binop);
+
+        let op = logical!(And).map(|_| BinaryOp::And);
+        let logical_and = bitwise_or
+            .clone()
+            .foldl_with_state(op.then(bitwise_or).repeated(), binop);
+
+        let op = logical!(Or).map(|_| BinaryOp::Or);
```

Comment:
Try commenting these out individually. With all enabled, I cannot get the project to build.